### PR TITLE
`<Button>`-component

### PR DESCRIPTION
**Pull request by KATT**
- Nicer `<Button/>` component
- Uses `next/link` + `<a/>` if you supply a `href` otherwise `<button/>`
- See output https://staging-git-feature-button-calend-so.vercel.app/sandbox/Button
  - Do you see any issues with having this route there? Easier than adding storybook. Added a noindex meta tag
- Also changed the `event-types/index` to use a query param for deciding if modal is open or not

---

### CODE REVIEW

**Review by PeerRich — APPROVED** [review]
beautiful and functional component. great job!

File: pages/event-types/index.tsx [hunk 1]
```diff
@@ -67,14 +70,23 @@ export default function Availability({ user, types }) {
...
+        };
+        delete newQuery["new"];
+        if (!isOpen) {
+          router.push({ pathname: router.pathname, query: newQuery });
+        }
+      }}>
+      <Button className="mt-2" StartIcon={PlusIcon} href={{ query: { ...router.query, new: "1" } }}>
         New event type
-      </DialogTrigger>
+      </Button>
```
Comment (KATT) [inline pages/event-types/index.tsx:86]: For reviewers:

I changed the modal here to act on state from query params

Comment (KATT) [reply pages/event-types/index.tsx:88]: cc @baileypumfleet @emrysal 

you like this or prefer using `useState`?

**Review by PeerRich — APPROVED** [review]

**Review by emrysal — APPROVED** [review]

**Comment by KATT** [comment]
Will merge this on Monday -- too late on Friday to deploy to production! :)

**Pull request merged**